### PR TITLE
Improve make and development setup

### PR DESCRIPTION
**Pull request by MarcelMue**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -22,45 +22,38 @@ jobs:
...
-            ./kubectl create -f https://raw.githubusercontent.com/kubernetes-sigs/cluster-api/release-0.3/config/crd/bases/cluster.x-k8s.io_clusters.yaml > /dev/null
-            ./kubectl create -f https://raw.githubusercontent.com/kubernetes-sigs/cluster-api/release-0.3/config/crd/bases/cluster.x-k8s.io_machinedeployments.yaml > /dev/null
+            make setup
       - architect/run-tests-with-abs:
           chart_dir: "./helm/policies-common"
           app-build-suite_version: "v0.2.3"
           app-build-suite_container_tag: "0.2.3"
           additional_app-build-suite_flags: "--external-cluster-version $KUBERNETES_VERSION"
+      - run:
+          name: Export kind logs
```
Comment (MarcelMue) [inline .circleci/config.yml:51]: Export the logs so we have all the pod logs in circle.

File: .circleci/config.yml [hunk 2]
```diff
@@ -22,45 +22,38 @@ jobs:
...
           command: |
-            ./kubectl create -f https://raw.githubusercontent.com/giantswarm/apiextensions/b9a6bed05850045530eaf8ba7fd01941de6c8525/helm/crds-common/templates/giantswarm.yaml > /dev/null
+            sudo apt-get update && sudo apt-get install make
       - run:
-          name: Install CAPI Cluster CRD
+          name: Setup
           command: |
-            ./kubectl create -f https://raw.githubusercontent.com/kubernetes-sigs/cluster-api/release-0.3/config/crd/bases/cluster.x-k8s.io_clusters.yaml > /dev/null
-            ./kubectl create -f https://raw.githubusercontent.com/kubernetes-sigs/cluster-api/release-0.3/config/crd/bases/cluster.x-k8s.io_machinedeployments.yaml > /dev/null
+            make setup
```
Comment (MarcelMue) [inline .circleci/config.yml:44]: Use make for local dev and ci now.

File: Makefile.custom.mk [hunk 3]
```diff
@@ -1,13 +1,17 @@
 .PHONY: generate
 generate:
-	./template.sh
+	./hack/template.sh
```
Comment (MarcelMue) [inline Makefile.custom.mk:3]: Move scripts to the `hack` folder to clean up a little bit.

File: hack/setup-local.sh [hunk 4]
```diff
@@ -1,8 +1,9 @@
...
+kind get kubeconfig --name kyverno-cluster > $(pwd)/kube.config
 kubectl create --context kind-kyverno-cluster -f https://raw.githubusercontent.com/giantswarm/apiextensions/b9a6bed05850045530eaf8ba7fd01941de6c8525/helm/crds-common/templates/giantswarm.yaml
 kubectl create --context kind-kyverno-cluster -f https://raw.githubusercontent.com/kubernetes-sigs/cluster-api/release-0.3/config/crd/bases/cluster.x-k8s.io_clusters.yaml
 kubectl create --context kind-kyverno-cluster -f https://raw.githubusercontent.com/kubernetes-sigs/cluster-api/release-0.3/config/crd/bases/cluster.x-k8s.io_machinedeployments.yaml
 kubectl create --context kind-kyverno-cluster -f https://raw.githubusercontent.com/kubernetes-sigs/cluster-api-provider-aws/main/config/crd/bases/infrastructure.cluster.x-k8s.io_awsclusters.yaml
 kubectl create --context kind-kyverno-cluster -f https://raw.githubusercontent.com/kubernetes-sigs/cluster-api-provider-aws/main/config/crd/bases/infrastructure.cluster.x-k8s.io_awsclusterroleidentities.yaml
+kubectl create --context kind-kyverno-cluster -f https://raw.githubusercontent.com/kyverno/kyverno/main/definitions/release/install.yaml
```
Comment (MarcelMue) [inline hack/setup-local.sh:9]: This needed to be reordered following upstream issue: https://github.com/kyverno/kyverno/issues/2094

File: helm/policies-aws/templates/AWSCAPI.yaml [hunk 5]
```diff
@@ -34,9 +34,6 @@ spec:
       - key: "{{ `{{` }} releaseVersion {{ `}}` }}"
         operator: Equals
         value: "20.0.0"
-      - key: "{{ `{{` }} request.object.metadata.labels.\"release.giantswarm.io/version\" {{ `}}` }}"
```
Comment (MarcelMue) [inline helm/policies-aws/templates/AWSCAPI.yaml:37]: Unrelated change which I messed up on main.

File: .circleci/config.yml (hunk 2 above)
Comment (stone-z) [reply .circleci/config.yml:44]: Neat

**Review by ericgraf — APPROVED** [review]

File: hack/setup-local.sh (hunk 4 above)
Comment (stone-z) [reply hack/setup-local.sh:9]: Does `abs` have logic to wait for kyverno to be up? If not, you could wait here. Something like https://github.com/giantswarm/coredns-warnlist-plugin/blob/f1d8ff108254e4f07968b718e933593de97ecd5d/.circleci/config.yml#L98

**Review by stone-z — APPROVED** [review]
One comment, but looking cleaner already

File: .circleci/config.yml [hunk 6]
```diff
@@ -22,45 +22,38 @@ jobs:
   test-policies:
     machine: true
     environment:
-      KIND_VERSION: v0.10.0
-      KUBERNETES_VERSION: v1.19.7
+      KIND_VERSION: v0.11.1
+      KUBERNETES_VERSION: v1.21.1
```
Comment (ubergesundheit) [inline .circleci/config.yml:26]: You'll want to update the kubernetes version also in the `.abs/main.yaml`

File: hack/setup-local.sh [hunk 7]
```diff
@@ -1,8 +1,9 @@
 kind create cluster --name kyverno-cluster
```
Comment (ubergesundheit) [inline hack/setup-local.sh:1]: This is missing the `--image` flag from ci

File: hack/setup-local.sh (hunk 4 above)
Comment (ubergesundheit) [reply hack/setup-local.sh:9]: app-build-suite (soon to be replaced with app-test-suite for testing) usually waits until the app cr shows status deployed

https://github.com/giantswarm/app-test-suite/blob/master/app_test_suite/steps/base_test_runner.py#L315

Comment (MarcelMue) [reply hack/setup-local.sh:9]: I thought about waiting here but currently it will just make the test run longer - kyverno is up way before the tests start. 

The problem is that kyverno is not the app-cr and therefor we wait for the setup of the kyverno-policies.

All in all not a big deal currently.

Comment (MarcelMue) [reply hack/setup-local.sh:9]: The problem with the bug I discovered was, that kyverno does currently not reload CRDs after it is started. This caused kyverno to fail if it was ready before all the CRDs were applied.

File: .circleci/config.yml (hunk 6 above)
Comment (MarcelMue) [reply .circleci/config.yml:26]: Done.

File: hack/setup-local.sh (hunk 7 above)
Comment (MarcelMue) [reply hack/setup-local.sh]: This should also be fixed now.

File: .circleci/config.yml (hunk 6 above)
Comment (MarcelMue) [reply .circleci/config.yml:26]: I am still looking for a way to have a single source of truth for this - currently thinking the make file / env and then replacing in `main.yaml`. 

Nothing seems beautiful though :sweat_smile:
